.github/workflows/diff.yml

```diff
@@ -1,13 +1,40 @@
 name: 'Create ROBOT diffs on Pull requests'
 
-on: 
+on:
 
   issue_comment:
         types: [created]
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
+  branch_status:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Check current branch status
+        id: info
+        run: |
+          echo "status=$(git rev-list --left-right --count origin/$DEFAULT_BRANCH...origin/${GITHUB_HEAD_REF} | awk '{print $1}')"
+          echo "status=$(git rev-list --left-right --count origin/$DEFAULT_BRANCH...origin/${GITHUB_HEAD_REF} | awk '{print $1}')" >> $GITHUB_OUTPUT
+      - name: Warn users with a comment
+        if: steps.info.outputs.status >= 1
+        run: echo "Your branch is $STATUS commit/s behind, please update your branch.   " > warncomment.md
+      - name: post comment
+        if: steps.info.outputs.status >= 1
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+        uses: NejcZdovc/comment-pr@v1.1.1
+        with:
+          file: "../../warncomment.md"
+      - name: fail
+        if: steps.info.outputs.status >= 1
+        run: |
+          echo "Your branch is $STATUS commit/s behind, please update your branch.   "
+          exit 1
```

Comment:
does this actually hard fail? I think I tried this once with another strategy and it still continues. But I guess you've tested it :)

Reply:
Good point @shawntanzk I have assumed this was tested, but yea better be sure

Reply:
@shawntanzk unless you use `continue-on-error: true` or `exit 0` it interrupts the run.